gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/ConnectionPool.java

```diff
@@ -665,13 +665,22 @@ private void appendConnections(final StringBuilder sb, final Connection connecti
                                    final List<Connection> connections) {
         final int connectionCount = connections.size();
         for (int ix = 0; ix < connectionCount; ix++) {
+            try {
             final Connection c = connections.get(ix);
-            if (c.equals(connectionToCallout))
+                if (c.equals(connectionToCallout)) {
                 sb.append("==> ");
-            else
+                }
+                else {
                 sb.append("> ");
-
+                }
             sb.append(c.getConnectionInfo(false));
+            } catch (final ArrayIndexOutOfBoundsException ex) {
```

Comment:
i mentioned this to @upadhyay-prashant elsewhere but others looking at this, i think we want to try to handle this differently as missing logs in the event of failure might prove to be confusing when debugging.